crates/ruff_linter/src/rules/ruff/rules/unnecessary_round.rs

```diff
@@ -0,0 +1,196 @@
+use ruff_python_semantic::analyze::typing;
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for `round()` calls that have no effect on the input.
+///
+/// ## Why is this bad?
+/// Rounding a value that's already an integer is unnecessary.
+/// It's clearer to use the value directly.
+///
+/// ## Example
+///
+/// ```python
+/// a = round(1, 0)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// a = 1
+/// ```
+#[derive(ViolationMetadata)]
+pub(crate) struct UnnecessaryRound;
+
+impl AlwaysFixableViolation for UnnecessaryRound {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Value being rounded is already an integer".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove unnecessary `round` call".to_string()
+    }
+}
+
+/// RUF057
+pub(crate) fn unnecessary_round(checker: &mut Checker, call: &ExprCall) {
+    let arguments = &call.arguments;
+
+    if !checker.semantic().match_builtin_expr(&call.func, "round") {
+        return;
+    }
+
+    let Some((rounded, rounded_value, ndigits_value)) = rounded_and_ndigits(checker, arguments)
+    else {
+        return;
+    };
+
+    if matches!(ndigits_value, NdigitsValue::Other) {
+        return;
+    }
+
+    let applicability = match rounded_value {
+        // ```python
+        // some_int: int
+        //
+        // rounded(1)
+        // rounded(1, None)
+        // rounded(1, 42)
+        // rounded(1, 4 + 2)
+        // rounded(1, some_int)
+        // ```
+        RoundedValue::Int(InferredType::Equivalent) => Applicability::Safe,
+
+        // ```python
+        // some_int: int
+        // some_other_int: int
+        //
+        // rounded(some_int)
+        // rounded(some_int, None)
+        // rounded(some_int, 42)
+        // rounded(some_int, 4 + 2)
+        // rounded(some_int, some_other_int)
+        // ```
+        RoundedValue::Int(InferredType::AssignableTo) => Applicability::Unsafe,
+
+        _ => return,
+    };
+
+    let edit = unwrap_round_call(checker, call, rounded);
+    let fix = Fix::applicable_edit(edit, applicability);
+
+    let diagnostic = Diagnostic::new(UnnecessaryRound, call.range);
+
+    checker.diagnostics.push(diagnostic.with_fix(fix));
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub(super) enum InferredType {
+    /// The value is an exact instance of the type in question.
+    Equivalent,
+    /// The value is an instance of the type in question or a subtype thereof.
+    AssignableTo,
+}
+
+/// The type of the first argument to `round()`
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub(super) enum RoundedValue {
+    Int(InferredType),
+    Float(InferredType),
+    Other,
+}
+
+/// The type of the second argument to `round()`
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub(super) enum NdigitsValue {
+    NotGiven,
+    LiteralNone,
+    Int(InferredType),
+    Other,
+}
+
+type Rounded = Expr;
```

Comment:
I removed the type alias because it wasn't used consistently and I found it more confusing than helpful (I had to lookup what its type is). In this case, I think adding some documentation might be more helpful